### ISSUE TITLE
Internal tool's Linux binary issues

### ISSUE THREAD

**Issue by datdo-msft** [microsoft/sbom-tool#272]
Hello,

We are from the AzureML team and have been using the new Linux binary of the _internal_ SBOM tool.

At first, we encountered this issue when we tried to run the tool:
```
Process terminated. Couldn't find a valid ICU package installed on the system. Please install libicu using your package manager and try again. Alternatively you can set the configuration flag System.Globalization.Invariant to true if you want to run with no globalization support. Please see https://aka.ms/dotnet-missing-libicu for more information.
   at System.Environment.FailFast(System.String)
   at System.Globalization.GlobalizationMode+Settings..cctor()
   at System.Globalization.CultureData.CreateCultureWithInvariantData()
   at System.Globalization.CultureData.get_Invariant()
   at System.Globalization.CultureInfo..cctor()
   at System.Globalization.CultureInfo.get_CurrentCulture()
   at System.String.EndsWith(System.String, System.StringComparison)
   at System.String.EndsWith(System.String)
   at PowerArgs.CommandLineAction.IsActionImplementation(System.Reflection.PropertyInfo)
   at PowerArgs.CommandLineArgument.IsArgument(System.Reflection.PropertyInfo)
   at PowerArgs.CommandLineArgumentsDefinition+<>c.<FindCommandLineArguments>b__97_1(System.Reflection.PropertyInfo)
   at System.Linq.Enumerable+WhereSelectArrayIterator`2[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].ToList()
   at System.Linq.Enumerable.ToList[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.Collections.Generic.IEnumerable`1<System.__Canon>)
   at PowerArgs.CommandLineArgumentsDefinition.FindCommandLineArguments(System.Type)
   at PowerArgs.CommandLineArgumentsDefinition..ctor(System.Type)
   at PowerArgs.Args.InvokeAction[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.String[])
   at PowerArgs.Args+<>c__DisplayClass32_0`1[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].<InvokeActionAsync>b__0()
   at System.Threading.Tasks.Task`1[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].InnerInvoke()
   at System.Threading.Tasks.Task+<>c.<.cctor>b__272_0(System.Object)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(System.Threading.Thread, System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef, System.Threading.Thread)
   at System.Threading.Tasks.Task.ExecuteEntryUnsafe(System.Threading.Thread)
   at System.Threading.Tasks.Task.ExecuteFromThreadPool(System.Threading.Thread)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()
   at System.Threading.Thread.StartCallback()
Aborted
```
We were able to work around this by following the guide here: https://learn.microsoft.com/en-us/dotnet/core/runtime-config/globalization and set the environment variable DOTNET_SYSTEM_GLOBALIZATION_INVARIANT to 1 (ie, running `export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1` before running the binary). Please let us know if this is recommended or not.

After that, we ran into another issue where we saw this:
```
Failure processing application bundle.
Failed to determine location for extracting embedded files.
DOTNET_BUNDLE_EXTRACT_BASE_DIR is not set, and a read-write cache directory couldn't be created.
```

and based on this answer here: [Azure Devops Pipeline, DOTNET_BUNDLE_EXTRACT_BASE_DIR is not set, and a read-write cache directory couldn't be created - Stack Overflow](https://stackoverflow.com/questions/74831700/azure-devops-pipeline-dotnet-bundle-extract-base-dir-is-not-set-and-a-read-wri), we created a directory for extracting these files and set the environment variable DOTNET_BUNDLE_EXTRACT_BASE_DIR to that directory (ie, running `export DOTNET_BUNDLE_EXTRACT_BASE_DIR="dir/to/extract/bundle/.net"` before running the native binary). This was able to unblock us and the tool successfully generated all the SBOM's. However, we saw that the files which were extracted to DOTNET_BUNDLE_EXTRACT_BASE_DIR did take up some space (~90MB):
![image](https://github.com/microsoft/sbom-tool/assets/131494842/b8dbc336-1a83-4c0c-980e-49b66f2a0b0a)

We are wondering if these are all necessary? Due to our use cases, we cannot introduce extra artifacts such as these extracted files.
